src/snake.py

```diff
@@ -1,31 +1,41 @@
 from const import *
+from vector import Vector, ZERO, RIGHT, LEFT, UP, DOWN
+
 
 class Snake:
-    
+
     def __init__(self):
         self.tail_count = 2
         self.locationX = 100
         self.locationY = 100
-        self.prev_locX = [80,60]
-        self.prev_locY = [80,60]
+        self.prev_locX = [80, 60]
+        self.prev_locY = [80, 60]
+        # now snake's direction is abstracted - it doesn't need to know anything about your input system
+        self.direction = DOWN
+
+    def change_direction(self, new_direction):
+        merged_direction = new_direction + self.direction
+        if merged_direction == ZERO:
+            return
```

Comment:
this hides logic a little - but if old input + new input == zero it means you tried to turn around which is a no no in snake